### PR TITLE
feat: Pricing calculator

### PR DESCRIPTION
**Pull request by maciaszczykm**
Besides pricing calculator it also adds required components like select item (to pick provider) and slider.

Zrzut ekranu 2023-03-16 o 11 59 38
Zrzut ekranu 2023-03-16 o 11 59 51
Zrzut ekranu 2023-03-16 o 12 00 10

Design: https://www.figma.com/file/vw6W6kaN658dkOAl74l3Hg/Pricing-Calculator?node-id=1-2&t=m5oeFv7TNUU3CdaD-0

### CODE REVIEW

**Comment by swoodward90** [comment]
@maciaszczykm we need to see how the meeting goes today, but I'm fairly certain we'll be implementing a cap on the number of users that an account can have on the free plan (see https://github.com/pluralsh/plural/pull/1007)
In that case, we'll need to adjust the extended slider to automatically flip to the Professional plan when > 5 users are selected and probably have an info callout on the users number the same way we do for Cloud Costs

**Review by floreks — APPROVED** [review]

File: src/components/pricingcalculator/controls/UsersControl.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+export default function UsersControl({
+  users,
+  setUsers,
+}: UsersControlProps) {
+  return (
+    <Control
+      header="How many users will access your clusters?"
+      caption="Plural is free for up to 5 open-source users. Account requiring more than 5 users and accounts on the professional plan will add $49/user."
```
Comment (dogmar) [inline src/components/pricingcalculator/controls/UsersControl.tsx:39]:
```suggestion
      caption="Plural is free for up to 5 open-source users. Accounts requiring more than 5 users and accounts on the professional plan will add $49/user."
```

**Review by dogmar — CHANGES_REQUESTED** [review]

**Comment by dogmar** [comment]
Looks good. Just caught one typo in some caption text.

**Comment by maciaszczykm** [comment]
@dogmar Fixed, PTAL.

**Pull request merged**